packages/hardhat/test/integration-tests/periphery/controller-helper.ts

```diff
@@ -0,0 +1,596 @@
+      collateralAmount = debtInEth.mul(3).div(2).add(ethers.utils.parseUnits('0.01'))
+      await controller.connect(tester).mintWPowerPerpAmount(0, mintWSqueethAmount, 0, {value: collateralAmount})
+      expect((await wSqueeth.balanceOf(tester.address)).gte(mintWSqueethAmount)).to.be.true
+    })
+
+    it("Sell long and flashswap mint short positon", async () => {
+      const longBalance = await wSqueeth.balanceOf(depositor.address);
+      const vaultId = (await shortSqueeth.nextId()).sub(1);
+      const normFactor = await controller.normalizationFactor()
+      const mintWSqueethAmount = ethers.utils.parseUnits('20')
+      const mintRSqueethAmount = mintWSqueethAmount.mul(normFactor).div(one)
+      const ethPrice = await oracle.getTwap(ethDaiPool.address, weth.address, dai.address, 420, true)
+      const scaledEthPrice = ethPrice.div(10000)
+      const debtInEth = mintRSqueethAmount.mul(scaledEthPrice).div(one)
+      const collateralAmount = debtInEth.mul(3).div(2).add(ethers.utils.parseUnits('0.01'))
+     
+      const swapParam = {
+        tokenIn: wSqueeth.address,
+        tokenOut: weth.address,
+        fee: 3000,
+        recipient: owner.address,
+        deadline: Math.floor(await getNow(ethers.provider) + 8640000),
+        amountIn: longBalance,
+        amountOutMinimum: 0,
+        sqrtPriceLimitX96: 0
+      }    
+      await wSqueeth.connect(depositor).approve(swapRouter.address, constants.MaxUint256)
+      const ethAmountOutFromSwap = await swapRouter.connect(depositor).callStatic.exactInputSingle(swapParam)
+
+      const flashswapParam = {
+        tokenIn: wSqueeth.address,
+        tokenOut: weth.address,
+        fee: 3000,
+        recipient: owner.address,
+        deadline: Math.floor(await getNow(ethers.provider) + 8640000),
+        amountIn: await wSqueeth.balanceOf(tester.address),
+        amountOutMinimum: 0,
+        sqrtPriceLimitX96: 0
+      }    
+      await wSqueeth.connect(tester).approve(swapRouter.address, constants.MaxUint256)
+      const ethAmountOutFromFlashSwap = await swapRouter.connect(tester).callStatic.exactInputSingle(flashswapParam)
+
+      const slippage = BigNumber.from(3).mul(BigNumber.from(10).pow(16))
+      const value = collateralAmount.sub(ethAmountOutFromSwap.mul(one.sub(slippage)).div(one)).sub(ethAmountOutFromFlashSwap.mul(one.sub(slippage)).div(one))
+      
+      await wSqueeth.connect(depositor).approve(controllerHelper.address, longBalance)
+      await controllerHelper.connect(depositor).flashswapSellLongWMint(0, mintWSqueethAmount, collateralAmount, longBalance, 0, {value: value})
+
+      const vaultAfter = await controller.vaults(vaultId)
+
+      expect((await wSqueeth.balanceOf(depositor.address)).eq(BigNumber.from(0))).to.be.true;
+      expect(vaultAfter.shortAmount.eq(mintWSqueethAmount)).to.be.true
+    })
+  })
+
+  describe("Close position with user wallet NFT: LP wPowerPerp amount is less than vault short amount", async () => {
```

Comment:
isnt in this case the wPowerPerp amount is = to the vault short amount (outside of rounding)?